src/ONYXKEYS.js

```diff
@@ -23,6 +23,9 @@ export default {
     // Credentials to authenticate the user
     CREDENTIALS: 'credentials',
 
+    // IOU Keys
```

Comment:
> Do we know how all the other IOUs will be stored in Onyx yet?

Here's the plan, but open to suggestions:
- **IOU** (generic keys used by IOUModal, IOUPreview, IOUDetail components):  `IOU { isLoading, isError, isComplete, participants}`
- **IOU Reports** (many individual reports, similar to reports): `COLLECTION.iouReport_`
- **IOU Report Transactions (similar to REPORT_ACTIONS)** `COLLECTION.reportTransactions_`
- **User preferred currency**: `MY_PERSONAL_DETAILS {preferredCurrency}`

Reply:
OK, thanks! One idea might be to move the `IOU` data into `APP.IOU` so that we know its data specifically related to the application and not data related to the IOU reports.

Reply:
Sounds good to me. The one part I'm a little skeptical about is exiting the Modal via `IOU { isComplete }`, does that seem like a reasonable way to confirm a successful IOUReport creation, so that we can close the Modal?

Reply:
hmmm, so for `isComplete`, that should really just be the output of something else, right? Like, it's complete when `all the information has been filled out`. At that point, the report is created, and the modal can be closed (via redirecting to a different route). I don't think I see the need for `isComplete` to be stored in Onyx.

Reply:
Implemented this for the `APP.IOU { loading: true}` key [here](https://github.com/Expensify/Expensify.cash/pull/1561/commits/edde0eb89b2843ced2c6817ab8545135f0d320ba). 

My only remaining concern is that `appIOU` isn't a great prop name. Maybe something like `appStateIOU` or `viewStateIOU` would be better?  perhaps `APP_STATE.IOU`, `VIEW.IOU`, or `COMPONENT.IOU`...

Reply:
In this case I'm thinking of creating the report and awaiting the successful API response (which occurs at the third step).

Reply:
I like `APP_STATE.IOU` to keep with the variable naming consistency in this file.

Reply:
Switched to `APP_STATE.IOU`

Reply:
Sorry, gonna maybe throw a wrench in this, but here goes... 

I'm not sure there is a clear case for dissociating the `loading` state from generic app data yet. Sounds like we are assuming that:

1. Multiple UI may use this data to display some feedback about loading (not sure it will ever be true)
2. It's valuable to have a concept of "app state" instead of handling network and client state in the same object

1 might not ever happen and 2 is a pretty big concept to introduce without more discussion in this PR.

So for this reason, my advice would be to:

1. Treat this `loading` state like we do others for now and just stick it on an object with the data.
2. Discuss whether we even need the distinction for "network state" (iou data returned from the server) vs. "client state" (local app UI loading stuff)
3. Depending on what we decide - update all the usages of the loading states and stick to that.

Reply:
Okay, fair points here -- I have concerns regarding shared loading indicators, and also the alternative of individual keys for every single indicator. Agree that this needs further discussion and I would prefer that this occurs outside of this PR, as this is blocking multiple issues and the solution will need a fair amount of consideration.

I'm going to move the loading key back to IOU for now.

Reply:
While we're here though, I wonder if IOU Onyx feature keys could remain unique and grouped like so:
```
IOU: {
        MODAL: 'iou_modal', `// {loading, error}`
        PREVIEW 'iou_preview', `// {loading}`
        DETAIL 'iou_detail', `// {loading}`
},
```